examples/chat-noise.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+extern crate env_logger;
+extern crate futures;
+extern crate libp2p;
+extern crate quicli;
+extern crate structopt;
+extern crate tokio;
+
+use futures::prelude::*;
+use libp2p::{
+    NetworkBehaviour, Transport, InboundUpgradeExt, OutboundUpgradeExt,
+    core::PublicKey,
+    tokio_codec::{FramedRead, LinesCodec}
+};
+use quicli::prelude::*;
+use structopt::StructOpt;
+use std::io;
+use tokio::runtime::Runtime;
+
+#[derive(Debug, StructOpt)]
+struct Cli {
+    #[structopt(long = "dial", short = "d")]
+    dial: Option<String>
+}
+
+fn main() -> CliResult {
```

Comment:
IMO a `chat-noise` example doesn't really make sense. If you want to test whether it works, I suggest an example or a test in `libp2p-noise`.
In the same vein, we're not going to create `chat-mplex`, `chat-yamux`, `chat-quic`, etc.
Integrating noise in the unique `chat` example does make sense to me however.